### ISSUE TITLE
Big difference between training accuracy and validation accuracy

### ISSUE THREAD

**Issue by Stuffooh** [jayleicn/TVQAplus#12]
Hi,

I'm trying to get a better understanding of the code and I'm running the following experiment:

1. Reproduce your results by simply running "bash run_main.sh --add_local". I managed to get around 72% validation score as expected.

2. I extract the features from a pretrained bert model for subtitles and QA pairs and then point the script to these new features and run "bash run_main.sh --add_local" again. I would expect again around 72% validation score as I extracted features from bert and saved them in the same h5 format. But instead I get a very high training accuracy (~90%) and a low validation accuracy (~52%) after around 30 epochs

I'm puzzled by these results. If the new extracted features were not good I would expect a low training and validation accuracy.  But instead the training accuracy is very high but validation accuracy is very low which would indicate over fitting or problem with the split of data between training and test  which both would be strange as it didn't happen originally and the code is not changed only the extracted features from bert changed.

I'm looking forward to some pointers what might be the cause of the big difference between training and validation accuracy for me.

**Comment by jayleicn** [comment]
Hi @Stuffooh,

Have you used early stop in the training?  The model reaches the best val accuracy at around 20 epoch. Besides, the performance for features without finetuning would be much lower than the finetuned version.

Jie

**Comment by Stuffooh** [comment]
Hi @jayleicn,

I am using early stop but I did no finetuning. I am trying to finetune now before extracting the features. Did you add any extra vocabulary before finetuning?

**Comment by jayleicn** [comment]
I used TVQA+ training set questions and correct answers, and their associated subtitles for this fine-tuning process. (validation and test should not be used).

I did not release the script for this fine-tuning, but I have released a similar fine-tuning script for the more recent TVR dataset, you may find some of the notes/scripts here to be useful: https://github.com/jayleicn/TVRetrieval/tree/master/utils/text_feature.

Discussions in this issue may also help: #6

**Comment by jayleicn** [comment]
> Hi @jayleicn,
> 
> I am using early stop but I did no finetuning. I am trying to finetune now before extracting the features. Did you add any extra vocabulary before finetuning?

I used the same set of vocabulary as in the hugging face BERT base uncased model, as it uses WordPiece tokenization, it should already cover a lot of words.

**Comment by Stuffooh** [comment]
Hi @jayleicn 

Thank you for all the extra in formation. I will look more into things and close this issue meanwhile ;)
